nsqlookupd/lookup_protocol_v1.go

```diff
@@ -191,6 +193,54 @@ func (p *LookupProtocolV1) UNREGISTER(client *ClientV1, reader *bufio.Reader, pa
 	return []byte("OK"), nil
 }
 
+func (p *LookupProtocolV1) SYNCSTATE(client *ClientV1, reader *bufio.Reader, params []string) ([]byte, error) {
+	if client.peerInfo == nil {
+		return nil, protocol.NewFatalClientErr(nil, "E_INVALID", "client must IDENTIFY")
+	}
+
+	topic, channel, err := getTopicChan("SYNCSTATE", params)
+	if err != nil {
+		return nil, err
+	}
+
+	var bodyLen int32
+	err = binary.Read(reader, binary.BigEndian, &bodyLen)
+	if err != nil {
+		return nil, protocol.NewFatalClientErr(err, "E_BAD_BODY", "SYNCSTATE failed to read body size")
+	}
+
+	body := make([]byte, bodyLen)
+	_, err = io.ReadFull(reader, body)
+	if err != nil {
+		return nil, protocol.NewFatalClientErr(err, "E_BAD_BODY", "SYNCSTATE failed to read body")
+	}
+
+	type bodyType struct {
+		Paused bool `json:"paused"`
+	}
+	var state bodyType
+	err = json.Unmarshal(body, &state)
+	if err != nil {
+		return nil, protocol.NewFatalClientErr(err, "E_INVALID", fmt.Sprintf("state body should be json"))
+	}
+
+	if channel != "" {
+		key := Registration{"channel", topic, channel}
+		if p.ctx.nsqlookupd.DB.UpdateProducer(key, client.peerInfo.id, state.Paused) {
```

Comment:
I don't think we need the semantics here where `UpdateProducer` returns a `bool` indicating whether there was an actual state change. In other words, we can always set the value to the current state and log.

Reply:
done.